lib/Runtime/Library/IntlEngineInterfaceExtensionObject.cpp

```diff
@@ -1087,11 +1132,13 @@ DEFINE_ISXLOCALEAVAILABLE(PR, uloc)
         if (kind == LocaleDataKind::Collation)
         {
             ScopedUEnumeration collations(ucol_getKeywordValuesForLocale("collation", localeID, false, &status));
-            ICU_ASSERT(status, true);
+            int collationsCount = uenum_count(collations, &status);
+
+            // we expect collationsCount to have at least "standard" and "search" in it
+            ICU_ASSERT(status, collationsCount > 2);
```

Comment:
I don't if it is guaranteed that all locales will have 2 (unique to the locale) collation types. I think that in some cases the "search" might be aliased to the "default/standard" collation type. 
However, that said, there might be fallback to the root data well, so they might always have at least two, but based on falling back to root.

Reply:
While debugging, I printed out all of the collation values for a few different locales and they all had at least two; if they don't, then our logic below could try to create a JavascriptArray of -1 elements (only if it has 0 collation values, which maybe is fine?), which would just failfast elsewhere (I hope...)

The Intl spec says that every locale has SearchLocaleData and SortLocaleData and usage _must_ accept both search and sort for all locales, so I think the API was designed with the assumption that everyone would have both. It seems like a rare enough case, at least, that I would be interested in being alerted via a defensive failfast if at some point it doesn't hold.

Reply:
Thanks Jack. To clarify I mean that not every individual locale (in and of itself) would necessarily have multiple collation types. (Like for example "en" might not anything specific to that locale at all). 
Since there is fallback to the root data, the net effect is that yes I think everybody would have multiple.